### PR TITLE
Website publishing improvements

### PR DESCRIPTION
**Pull request by sbrudz**
This PR improves the publishing process for https://justnotsorry.com with the goal of making website updates easier and clearer.

Changes include:
- [x] Moved all the jekyll site generator code from the `gh-pages` branch into the `site` subdirectory
- [x] Configured gh-pages npm package to deploy the `site` subdirectory to the `gh-pages` branch
- [x] Configured a GitHub actions workflow that will trigger a site deployment when relevant files change
- [x] Added Dynamic generation of the "List of Warning Phrases" page content based on `phrases.json`
- [x] Added a git commit SHA and publication timestamp to the footer of justnotsorry.com
- [x] Updated ReadMe files to describe new process

### CODE REVIEW

**Comment by jduffy3** [comment]
Had an issue running bundle install locally on my machine:

```
An error occurred while installing eventmachine (1.2.7), and Bundler cannot continue.

In Gemfile:
  github-pages was resolved to 219, which depends on
    jekyll-avatar was resolved to 0.7.0, which depends on
      jekyll was resolved to 3.9.0, which depends on
        em-websocket was resolved to 0.5.3, which depends on
          eventmachine
```

 but was able to resolve it with this

https://stackoverflow.com/questions/74196882/cannot-install-jekyll-eventmachine-on-m1-mac

File: site/phrases.ejs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# List of Warning Phrases
```
Comment (jduffy3) [inline site/phrases.ejs:1]: Not sure if this file is still needed? Are we using `site/phrases.md` instead?

**Review by jduffy3 — APPROVED** [review]
Approving but had one question about a file not sure is required.

**Comment by sbrudz** [comment]
> Had an issue running bundle install locally on my machine:
> 
> ```
> An error occurred while installing eventmachine (1.2.7), and Bundler cannot continue.
> 
> In Gemfile:
>   github-pages was resolved to 219, which depends on
>     jekyll-avatar was resolved to 0.7.0, which depends on
>       jekyll was resolved to 3.9.0, which depends on
>         em-websocket was resolved to 0.5.3, which depends on
>           eventmachine
> ```
> 
> but was able to resolve it with this
> 
> https://stackoverflow.com/questions/74196882/cannot-install-jekyll-eventmachine-on-m1-mac

What version of ruby and openssl were you using?  I wonder if I should upgrade to ruby 3 since 2.7's EOL is in March.

File: site/phrases.ejs (hunk 1 above)
Comment (sbrudz) [reply site/phrases.ejs:1]: Good catch!  I should have deleted that ejs file.  I'll take care of it.  Thank you!

**Comment by jduffy3** [comment]
```
➜  just-not-sorry git:(website-publishing-improvements) ✗ openssl version
LibreSSL 3.3.6
```
I had to install 2.7 with rvm to start the `bundle install`
```
➜  site git:(website-publishing-improvements) ✗ ruby -v
ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [arm64-darwin22]
```

However I noticed that locally this updated `site/Gemfile.lock` locally for me with the following PLATFORM: `arm64-darwin-22`

File: site/phrases.ejs (hunk 1 above)
Comment (sbrudz) [reply site/phrases.ejs]: phrases.ejs file deleted.

**Comment by sbrudz** [comment]
I create a new issue for the ruby upgrade: https://github.com/defmethodinc/just-not-sorry/issues/171

**Comment by sbrudz** [comment]
> However I noticed that locally this updated `site/Gemfile.lock` locally for me with the following PLATFORM: `arm64-darwin-22`

This seems to be fine as the PLATFORM gets updated when `bundle install` is run on a new platform (per [this article](https://www.moncefbelyamani.com/understanding-the-gemfile-lock-file/#platforms)).  I'll bet you're on OS 13.  I'm still on 12.

**Comment by jduffy3** [comment]
> > However I noticed that locally this updated `site/Gemfile.lock` locally for me with the following PLATFORM: `arm64-darwin-22`
> 
> This seems to be fine as the PLATFORM gets updated when `bundle install` is run on a new platform (per [this article](https://www.moncefbelyamani.com/understanding-the-gemfile-lock-file/#platforms)). I'll bet you're on OS 13. I'm still on 12.

Yup I'm on 13!

**Comment by sbrudz** [comment]
@jduffy3 I added a job to the justnotsorry.com GitHub Action workflow that will run `bundle install` and perform a test build of the site.  It'll only publish once the changes are on the main branch.  This should help catch potential issues with the ruby/jekyll configuration during the PR stage.

**Pull request merged**